### PR TITLE
feat(integrations): [nan-1598] add cal-v2 syncs

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Adds `events` and `event-types` syncs

## Issue ticket number and link
NAN-1598

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is: 

## Summary by CodeRabbit

- **New Features**
	- Introduced synchronization capabilities for managing events and event types.
	- Added functionality to retrieve and process upcoming events and event types with scheduled sync operations.

- **Bug Fixes**
	- Enhanced data retrieval robustness with retry mechanisms for event and event type fetching.

- **Documentation**
	- Updated configuration files to include new synchronization settings and data models for better clarity and usage.

### CODE REVIEW

File: integration-templates/cal-com-v2/nango.yaml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            events:
+                sync_type: full
+                runs: every hour
+                endpoint: GET /events
+                output: Event
+                description: Retrieve all upcoming events per a user
+            event-types:
+                sync_type: full
+                runs: every hour
+                endpoint: GET /event/types
```
Comment (bodinsamuel) [inline integration-templates/cal-com-v2/nango.yaml:15]:
```suggestion
                endpoint: GET /events/types
```

File: integration-templates/cal-com-v2/types.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+export interface EventTypeGroup {
+    teamId: string | null;
+    bookerUrl: string;
+    membershipRole: string | null;
+    profile: {
+        slug: string;
+        name: string;
+        image: string;
+    };
+    eventTypes: EventType[];
```
Comment (bodinsamuel) [inline integration-templates/cal-com-v2/types.ts:19]: not sure its advisable to mix our own models and theirs

File: integration-templates/cal-com-v2/nango.yaml [hunk 3]
```diff
@@ -0,0 +1,163 @@
+integrations:
+    cal-com-v2:
+        syncs:
+            events:
+                sync_type: full
+                runs: every hour
+                endpoint: GET /events
+                output: Event
+                description: Retrieve all upcoming events per a user
```
Comment (bodinsamuel) [inline integration-templates/cal-com-v2/nango.yaml:10]: I always forget if it's worth setting track_deletes when sync_type:full

File: integration-templates/cal-com-v2/nango.yaml [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        bookerUrl: string
+    Attendee:
+        id: number
+        email: string
+        name: string
+        timeZone: string
+        locale: string
+        bookingId: number
+    Event:
+        id: number
```
Comment (bodinsamuel) [inline integration-templates/cal-com-v2/nango.yaml]: id should be string

File: integration-templates/cal-com-v2/nango.yaml [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+            event-types:
+                sync_type: full
+                runs: every hour
+                endpoint: GET /event/types
+                output: EventType
+                description: Retrieve all event types per a user
+
+models:
+    EventType:
+        id: number
```
Comment (bodinsamuel) [inline integration-templates/cal-com-v2/nango.yaml]: same

**Review by bodinsamuel — APPROVED** [review]

File: integration-templates/cal-com-v2/nango.yaml (hunk 4 above)
Comment (khaliqgant) [reply integration-templates/cal-com-v2/nango.yaml]: Yeah, for Nango purposes it should be a string -- updated

File: integration-templates/cal-com-v2/nango.yaml (hunk 5 above)
Comment (khaliqgant) [reply integration-templates/cal-com-v2/nango.yaml]: Updated

File: integration-templates/cal-com-v2/nango.yaml (hunk 3 above)
Comment (khaliqgant) [reply integration-templates/cal-com-v2/nango.yaml:10]: It is. Thanks, updated

File: integration-templates/cal-com-v2/types.ts (hunk 2 above)
Comment (khaliqgant) [reply integration-templates/cal-com-v2/types.ts:19]: Will keep as is and can refine if needed later. This is for a customer, so assume they'll want a tighter model but they haven't let us know what that is yet.

File: integration-templates/cal-com-v2/nango.yaml (hunk 1 above)
Comment (khaliqgant) [reply integration-templates/cal-com-v2/nango.yaml:15]: Will keep as is as it is matching the calendly endpoint.

**Pull request merged**